File: leetcode/2798-number-of-employees-who-met-the-target/numberOfEmployeesWhoMetTarget_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type NumberOfEmployeesWhoMetTargetCase struct {
	Input struct {
		Hours  []int
		Target int
	}

	Output int
}

func TestNumberOfEmployeesWhoMetTarget(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]NumberOfEmployeesWhoMetTargetCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)
		fmt.Printf("string: %v ==> %v \n", testCase.Input, testCase.Output)

		result := numberOfEmployeesWhoMetTarget(testCase.Input.Hours, testCase.Input.Target)
		if result != testCase.Output {
			t.Errorf("期望 %v, 结果 %v", testCase.Output, result)
		}
	}
}

File: leetcode/2798-number-of-employees-who-met-the-target/numberOfEmployeesWhoMetTarget.test.ts
import cases from "./cases.json";
import numberOfEmployeesWhoMetTarget from "./numberOfEmployeesWhoMetTarget";

test("number of employees who met the target", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    const result = numberOfEmployeesWhoMetTarget(
      testCase.input.hours,
      testCase.input.target,
    );
    console.log(
      `input: ${testCase.input} , output: ${testCase.output} ==> ${result} \n`
    );

    expect(result).toEqual(testCase.output);
  });
});
